lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,203 @@
+            // Libraries SHOULD not allow to send id as additional field ( _id). Graylog server nodes omit this field automatically.
+            .unknown_fields(Kind::bytes())
+        //
+    }
+}
+
+/// Deserializer that builds an `Event` from a byte frame containing a GELF log
+/// message.
+#[derive(Debug, Clone)]
+pub struct GelfDeserializer {
+    regex: Regex,
+}
+
+impl Default for GelfDeserializer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl GelfDeserializer {
+    /// Create a new GelfDeserializer
+    pub fn new() -> GelfDeserializer {
+        GelfDeserializer {
+            regex: Regex::new(r"^[\w\.\-]*$").unwrap(),
+        }
+    }
+
+    // Adds fields from parsed GelfMessage to the event, adhering to GELF spec
+    fn insert_fields_from_gelf(
+        &self,
+        event: &mut Event,
+        parsed: &GelfMessage,
+    ) -> vector_core::Result<()> {
+        let log = event.as_mut_log();
+
+        log.insert(VERSION, parsed.version.to_string());
+        log.insert(HOST, parsed.host.to_string());
+
+        if let Some(full_message) = &parsed.full_message {
+            log.insert(FULL_MESSAGE, full_message.to_string());
+        }
+
+        if let Some(timestamp) = parsed.timestamp {
+            let naive = NaiveDateTime::from_timestamp(
+                f64::trunc(timestamp) as i64,
+                f64::fract(timestamp) as u32,
+            );
+            log.insert(
+                log_schema().timestamp_key(),
+                DateTime::<Utc>::from_utc(naive, Utc),
+            );
+        } else {
+            log.insert(log_schema().timestamp_key(), Utc::now());
+        }
+
+        if let Some(level) = parsed.level {
+            log.insert(LEVEL, level);
+        }
+        if let Some(facility) = &parsed.facility {
+            log.insert(FACILITY, facility.to_string());
+        }
+        if let Some(line) = &parsed.line {
+            log.insert(LINE, *line);
+        }
+        if let Some(file) = &parsed.file {
+            log.insert(FILE, file.to_string());
+        }
+
+        if let Some(add) = &parsed.additional_fields {
+            for (key, val) in add.iter() {
+                // per GELF spec, filter out _id
+                if key == "_id" {
+                    continue;
+                }
+                // per GELF spec, Additional field names must be characters dashes or dots
+
+                // TODO currently dropping fields that don't match the GELF spec... should we error instead?
+                if !self.regex.is_match(key) {
+                    continue;
+                }
+                // TODO Vector seems to not accept "." or "-" for log field names...
+                // not sure if this is the best approach but seems either we will need to modify the
+                // field names we received, or alter vector to allow "." or "-" in the field names.
```

Comment:
Unsure if this is true, or just from the decoding side of things. For example the ec2 metadata transform adds fields with `-` in them.

Reply:
Hmm. I can try again but when I ran my tests, if I didn't have this code, then the fields containing those characters were not in the resulting event (so my unit test failed)

Reply:
Yeah, that is correct, you should be able to insert them. Users will just need to quote them to access them. I think the below might need to be:

```
log.insert(path!(key.as_str()), val.as_str());
```

Reply:
So after I added the path!() macro , I am seeing those fields in the event << when in the parse() function >> , but the unit test fails :) so something after the `parse()` is removing it ... I'll try to debug that

Reply:
scratch that! I had to add the `path!()` macro in the unit test also. Works great now, good catch!